src/effects/LensFlare.tsx

```diff
@@ -506,118 +505,117 @@ type LensFlareProps = {
-      animated,
-      anamorphic,
-      colorGain,
-      lensDirtTexture,
-      haloScale,
-      secondaryGhosts,
-      aditionalStreaks,
-      ghostScale,
-      opacity,
-      starBurst,
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    const effect = useMemo(() => new LensFlareEffect(opts), [JSON.stringify(opts)])
-
-    useFrame((_, delta) => {
-      const uLensPosition = effect.uniforms.get('lensPosition')
-      const uOpacity = effect.uniforms.get('opacity')
-      if (!uLensPosition || !uOpacity) return
-
-      let target = 1
-
-      if (followMouse) {
-        uLensPosition.value.x = pointer.x
-        uLensPosition.value.y = pointer.y
-        target = 0
-      } else {
-        projectedPosition.copy(position).project(camera)
-        if (projectedPosition.z > 1) return
-
-        uLensPosition.value.x = projectedPosition.x
-        uLensPosition.value.y = projectedPosition.y
-
-        mouse2d.set(projectedPosition.x, projectedPosition.y)
-        raycaster.setFromCamera(mouse2d, camera)
-        const intersects = raycaster.intersectObjects(scene.children, true)
-        const { object } = intersects[0]
-        if (object) {
-          if (object.userData?.lensflare === 'no-occlusion') {
-            target = 0
-          } else if (object instanceof THREE.Mesh) {
-            if (object.material.uniforms?._transmission?.value > 0.2) {
-              //Check for MeshTransmissionMaterial
-              target = 0.2
-            } else if (object.material._transmission && object.material._transmission > 0.2) {
-              //Check for MeshPhysicalMaterial with transmission setting
-              target = 0.2
-            } else if (object.material.transparent) {
-              // Check for OtherMaterials with transparent parameter
-              target = object.material.opacity
-            }
+const LensFlareWrapped = wrapEffect(LensFlareEffect)
+
+export const LensFlare = ({
+  position = new THREE.Vector3(-25, 6, -60),
+  followMouse = false,
+  smoothTime = 0.07,
+  //
+  blendFunction = BlendFunction.NORMAL,
+  enabled = true,
+  glareSize = 0.2,
+  lensPosition = new THREE.Vector2(0.01, 0.01),
+  screenRes = new THREE.Vector2(0, 0),
+  starPoints = 6,
+  flareSize = 0.01,
+  flareSpeed = 0.01,
+  flareShape = 0.01,
+  animated = true,
+  anamorphic = false,
+  colorGain = new THREE.Color(20, 20, 20),
+  lensDirtTexture = null,
+  haloScale = 0.5,
+  secondaryGhosts = true,
+  aditionalStreaks = true,
+  ghostScale = 0.0,
+  opacity = 1.0,
+  starBurst = false,
+}: LensFlareProps) => {
+  const viewport = useThree(({ viewport }) => viewport)
+  const raycaster = useThree(({ raycaster }) => raycaster)
+  const pointer = useThree(({ pointer }) => pointer)
+  const { scene, camera } = useContext(EffectComposerContext)
+
+  const [projectedPosition] = useState(() => new THREE.Vector3())
+  const [mouse2d] = useState(() => new THREE.Vector2())
+
+  const ref = useRef<LensFlareEffect>(null)
```

Comment:
You'll need a non-null assertion there to clear the earlier error.

```ts
const ref = useRef<LensFlareEffect>(null~)
```

Reply:
I've just tried in https://github.com/pmndrs/react-postprocessing/pull/231/commits/c9788b3d764d52d8b38710502aee993cebdd2d07 but still yelling:
```
Type 'MutableRefObject<LensFlareEffect>' is not assignable to type 'Ref<typeof LensFlareEffect> | undefined'.
  Type 'MutableRefObject<LensFlareEffect>' is not assignable to type 'RefObject<typeof LensFlareEffect>'.
    Types of property 'current' are incompatible.
      Property 'prototype' is missing in type 'LensFlareEffect' but required in type 'typeof LensFlareEffect'.ts(2322)
```